### ISSUE TITLE
[Installation] ERROR: Can't find Sketch with that signature. Hash is invalid.

### ISSUE THREAD

**Issue by laurapons** [duraki/SketchCrapp#28]
Hello,
I downloaded Sketch v69 from the website, and went through the steps mentioned in the readme but I got this error. I am working with a Macbook Pro with Catalina 10.15.7 installed.

I just opened Sketch and it says I have 1 day left :O
Is there anything I can do to solve it? Thanks in advance

SketchCrapp is finding application bundle path ...
[+] Selected Sketch.app path is </Applications> (auto-detected) ... OK
[+] Analysing application bundle ... Starting
[+] Finding executable file ... OK
[+] Finding Info.plist ... OK
[+] Checking Info.plist for CFBundleShortVersionString ... OK
[+] Validating executable file ... Error
[ERR] Can't find Sketch with that signature. Hash is invalid.
[INFO] Carefully review README file again
[INFO] If you still have problem copy the details below and open a new issue on GitHub repository: https://github.com/duraki/SketchCrapp
+===================================================================
+ Application Path  : /Applications/Sketch.app
+ Application Binary: /Applications/Sketch.app/Contents/MacOS/Sketch
+ Passed version    : 69
+ Correct hash      : 651f3263305e004133253c2706fcdf5b16e20558
+ Binary SHA1       : 63e50175576e15df17809d2a3f72a19539651bdc
+ Error             : Can't find Sketch with that signature. Hash is invalid.
+===================================================================

**Comment by elijahtsai** [comment]
record one: why is the version 69's hash is 68.2's hash?

**Comment by elijahtsai** [comment]
> record one: why is the version 69's hash is 68.2's hash?

Fixed in PR #29

**Comment by elijahtsai** [comment]
Hello @laurapons, First thanks for tried out our creation, according to the debug info you provided us, It seems like the application you unzipped from the zip file which you downloaded from the official website, the application had been modified before, **please make sure after you unzip do not open the application before the script successfully patched.**

**Comment by laurapons** [comment]
Good morning @elijahtsai 
I just tried deleting sketch.app form Applications as well as the sketchcrapp folder and repeat the installation process making sure I didn't launch Sketch before applying the patch.
1. Downloaded sketch-69-107357.zip, unzip and send to Applications (without opening).
2. Copy&paste commands on Terminal to install and launch sketchcrapp
3. This is the output of the script:

Lauras-MacBook-Pro:sketchcrapp laura$ **cd $HOME && git clone https://github.com/duraki/sketchcrapp**
Cloning into 'sketchcrapp'...
remote: Enumerating objects: 104, done.
remote: Counting objects: 100% (104/104), done.
remote: Compressing objects: 100% (78/78), done.
remote: Total 293 (delta 42), reused 71 (delta 24), pack-reused 189
Receiving objects: 100% (293/293), 82.56 KiB | 249.00 KiB/s, done.
Resolving deltas: 100% (97/97), done.
Lauras-MacBook-Pro:~ laura$ **cd $HOME/sketchcrapp && chmod +x sketchcrapp.sh**
Lauras-MacBook-Pro:sketchcrapp laura$ **cd $HOME/sketchcrapp && ./sketchcrapp.sh**
           __       __      __
      ___ / /_____ / /_____/ /  ___________ ____  ___
    ( _-</  '_/ -_) __/ __/ _ \/ __/ __/ _ `/ _ \/ _ \
    /___/_/\_\\__/\__/\__/_//_/\__/_/  \_,_/ .__/ .__/
                                          /_/  /_/
         Sketch.App Patch Tool (https://github.com/duraki/SketchCrapp)
         by @duraki & @elijahtsai

SketchCrapp is finding application bundle path ...
[+] Selected Sketch.app path is </Users/laura/Applications> (auto-detected) ... OK
[+] Analysing application bundle ... Starting
[+] Finding executable file ... OK
[+] Finding Info.plist ... OK
[+] Checking Info.plist for CFBundleShortVersionString ... OK
[+] Validating executable file ... OK
[+] Selected Sketch.app version is 69 ... SketchCrapp starting ... OK
[+] Patching offsets for 69 ... Starting
[+] Patching address at offset: 0x5cf76f with value: \00
1+0 records in
1+0 records out
1 bytes transferred in 0.000028 secs (35849 bytes/sec)
[+] Patching address at offset: 0x5cf772 with value: \00
1+0 records in
1+0 records out
1 bytes transferred in 0.000022 secs (45100 bytes/sec)
[+] Patching address at offset: 0x5ce44a with value: \00\00
2+0 records in
2+0 records out
2 bytes transferred in 0.000031 secs (64528 bytes/sec)
[+] Patching address at offset: 0x5ce589 with value: \165
1+0 records in
1+0 records out
1 bytes transferred in 0.000021 secs (47663 bytes/sec)
[+] SketchCrapp certificate already exists. Skipping certificate creation ... OK
[+] Signing the patched *.app bundle. This may require sudo.
[+] If asked, enter your login password. Choose "Always Allow" to not be asked again.
/Users/laura/Applications/Sketch.app: replacing existing signature
[+] Cleaning up file(s) ... Cleaned
[+] SketchCrapp process completed. Sketch.app has been patched :)
[+] -- Notice: 
[+] If a dialogue shows up with message: “Sketch 3.app” can’t be opened
[+] please right-click the application and select open, or go to Settings -› Security
[+] and allow opening Sketch.app application.

[+] SketchCrapp (A Sketch.app cracking tool)
[+] https://github.com/duraki/SketchCrapp [by @duraki & @elijahtsai]
Lauras-MacBook-Pro:sketchcrapp laura$ 

Still, I see the 1 day left:
<img width="340" alt="Captura de Pantalla 2020-10-19 a les 7 57 26" src="https://user-images.githubusercontent.com/1275184/96406989-bd3a6780-11e0-11eb-9d27-9f1eefebc241.png">

**Comment by duraki** [comment]
@laurapons 

It's normal to see "1 days left", the product will continue working forever. The patch completed successfully. In fact, we are working on hack that should hide that message because many users find it irritating, see #11 

Also see #8 

Thanks for the head-ups.

**Comment by laurapons** [comment]
Great news!! Thanks for your effort and for your quick reply. Closing the issue :)
